WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+import Yosemite
+import class WooFoundation.CurrencyFormatter
+
+class CardPresentPaymentsAdaptor {
+    private let currencyFormatter: CurrencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
+    private let siteID: Int64
+
+    init(siteID: Int64) {
+        self.siteID = siteID
+    }
+
+    /// The problem with this approach is that making the usecase also makes the preflight controller, which makes the two connection controllers.
```

Comment:
What are the reasons this adaptor can't keep the 2 connection controllers and then pass them in a preflight controller like in https://github.com/woocommerce/woocommerce-ios/pull/12735/files#diff-2c42212ff82dbd5a932472da2fdfa48bf6af9ca77d78c4d5ea7ad3985b47632cR53?

Reply:
Because the connection controllers would also need to have the `eventAdaptor`, which has to be short-lived as it is tied to the same lifecycle as the stream. 

Perhaps we could have a long-lived `eventAdaptor` which could be given a new `streamContinuation` whenever a new payment started. That seems like it would be quite error-prone, as streams may not be `finished` before they're replaced, and if we use streams I want to make it more difficult to misuse them than they are currently.